### DISCUSSION TITLE
Query on SI parameters

### DISCUSSION THREAD

**Discussion by SaiBhyri** [Telecominfraproject/oopt-gnpy#472] [Q&A] [answered]
Hi,

Have below queries on SI and Span parameters and need help to understand better:
 
**power_dbm** – Is it the per channel power coming out of the amplifier or per channel launch power into the fiber?
**power_range_db** – Is it the variation around power_dbm? For eg: If power_dBm is set to +1dBm and power_range_dB is set to [-1,+1,0.5] then would the simulation be performed for Power values = 0dBm, 0.5dBm, 1dBm, 1.5dBm and 2dBm?
What is **delta_power_range_db**? Sorry, I couldn’t understand well from the documentation.
There is reference to **delta_p** in above parameter definitions but couldn’t find its definition. What does it signify?
What is target_extended_gain? Its present in json definition but not in documentation.
 
 
**baud_rate** and **spacing** are present both under SI parameters and Transceiver parameters. How and where are they used?
I have noticed from my simulations that GSNR is independent of **baud_rate** and **spacing** defined under SI parameters. Is GSNR completely governed by **baud_rate** and **spacing** defined under Transceiver parameters?

Thanks,
Sai

**Comment by EstherLerouzic** [comment, marked as answer]
Dear SaiBhyri

SI contains the reference channel type used for network design (if no settings are set on amplifiers)
- gnpy-transmission-main also uses this reference channel for propagation simulation
- gnpy-path-request uses the transceiver's mode definition for propagation. So, in this case, the propagation is performed for channels according to Transceiver's mode. Performance for each service is computed on a worst case basis, assuming the whole spectrum is filled with this transceiver type_variety and mode.

If no settings are set on (some or all) amplifiers the simulator automatically selects one amplifier type_variety among the ones defined in the library that can support the total power target and gain. The total power target is computed for the target amount of channels of the reference type defined in SI. So each amplifier must support 
- (**f_max** - **f_min**)/(**spacing**) nb of channels,
-  each with **power_dbm** target power out.  

**power_dbm** is the target power per channel at the output of amplifiers. If a fiber is connected to this amp, then this corresponds to the per channel target power at the input of the fiber span = input power per span.

The optimum input power per span is not the same for all spans (depends on length and loss), so **delta_power_range_db** enables a range around **power_dbm** to optimize the input power per span. This deviation around power_dbm is recorded into **delta_p** in each amplifier instance in the topology. When set to [0, 0, xxx] no optimization is done, so the design strictly uses **power_dbm**. If the amplifier instance already contains a setting, no computation or optimization are applied and the user settings are applied unless they are not consistent (for example in case settings are not consistent with max power of amplifier type_variety).

If the amplifier maximum power can not support the computed amount of channels with this target power, a reduction is applied and recorded into **delta_p**.

**power_range_db** exactly does what you say, a power sweep: (after performing a design for the target channel with power **power_dbm**, if no settings are present in one or some amplifiers), it propagates a signal for the same type of channel but different target powers. Computed gains remain the ones of the design unless the new computed target_power per channel hits the max power of this amplifier type_variety. This feature enables to explore different target power and identify the best one (to maximize GSNR for example). In case amplifiers had non zero delta_p defined in their operationnal settings, these are also applied so that the whole link target powers are translated up or down depending on the range.

 Hope this helps,

regards

**Reply by SaiBhyri** [reply]
Thanks Esther for detailed response.

**Comment by SaiBhyri** [comment]
Hi Esther,

Need your help on further queries:

- If there is more than one amplifier satisfying total power target and gain, do we choose the one which gives best OSNR based on NF?

- While running gnpy-path-request and in the case of having more than one transceiver type, is the per channel launch power into the span the worst one among the per channel span launch power calculated for every transceiver mode?

- Similarly, for the above case, is the amplifier selection based on worst/most demanding total power target and gain (as total no. of channels will differ based on transceiver mode definition) among all the transceiver modes?

- Is the optimization of per span launch power done of get best GSNR from every ROADM-ROADM section? For eg: Consider a linear network : Site 1 (ROADM) <-> Site 2 (ILA) <-> Site 3 (ROADM) <-> Site 4 (ILA) <-> Site 5 (ROADM). There are 4 fiber spans and 2 ROADM-ROADM sections. Is the per span launch power optimization done independently on both ROADM-ROADM section?

- Is the per span launch power optimization logic following (considering above topology and with power_dBm = 0dBm and power_range_db = [-1,+1,0.5] ). 
           - Site 1 BST power vary from -1dBm to +1dBm in steps of 0.5dB keeping the BST output powers at Site2, Site 3 at 0dBm.
           - Choose the output power value at Site 1 BST which gives the best GSNR from above run.
           - Lock the Site 1 BST value to above found value. Vary Site 2 BST output power from -1dBm to +1dBm in steps of 0.5dB and 
              keep  BST output power at Site3 at 0dBm.
           - Choose the output power value at Site 2 BST which gives the best GSNR from above run.
           - Lock the Site 1 BST and Site 2 BST value to above found values. Vary Site 3 BST output power from -1dBm to +1dBm in steps 
              of 0.5dB.
           - In this manner, find the optimal launch power values for BSTs at Site 1, Site 2, Site 3.
           - Now come back to Site 1 BST and vary its power value from -1dBm to +1dBm in steps of 0.5dB keeping the BST amps output 
              power Site 2, Site 3 at their optimal power value found from above runs.
            - If Site 1 BST gives same output power value as before, then move to Site 2, Site 3 and check the same.
            - If above run gives the same BST out power values at Site 1, Site 2,Site 3 then convergence has happened else repeat all the 
               above runs till a convergence is reached

Thanks,
Sai

**Reply by SaiBhyri** [reply]
Hi Esther,

Could you please help on my further queries? 

Thanks,
Sai

**Comment by EstherLerouzic** [comment]
Dear sai,

here are some answers:

> * If there is more than one amplifier satisfying total power target and gain, do we choose the one which gives best OSNR based on NF?

yes, it selects the one with lowest NF

> * While running gnpy-path-request and in the case of having more than one transceiver type, is the per channel launch power into the span the worst one among the per channel span launch power calculated for every transceiver mode?

No, each request performance is computed independantly from the other request: the propagation is performed full load (according to request setting) for the same channel (all channels with the same power, baudrate, spacing, ... settings). If you want to simulate mixed channel, you must use transmission program with the -spectrum option.

> * Similarly, for the above case, is the amplifier selection based on worst/most demanding total power target and gain (as total no. of channels will differ based on transceiver mode definition) among all the transceiver modes?

Selection of amplifier is performed base on the reference channel defined in SI part of eqpt_config.json input. Then propagation for each request uses this very first design. so the amplifier type do not depend on the requests. The gain target or delta_p may however be reduced if the request specifies the channel type and number that leads to a total power that exceeds chosen amp capability (saturation).

> * Is the optimization of per span launch power done of get best GSNR from every ROADM-ROADM section? For eg: Consider a linear network : Site 1 (ROADM) <-> Site 2 (ILA) <-> Site 3 (ROADM) <-> Site 4 (ILA) <-> Site 5 (ROADM). There are 4 fiber spans and 2 ROADM-ROADM sections. Is the per span launch power optimization done independently on both ROADM-ROADM section?

optimum channel power is performed per OMS. so in your example Site2 settings is independant from Site1 settings. 

> * Is the per span launch power optimization logic following (considering above topology and with power_dBm = 0dBm and power_range_db = [-1,+1,0.5] ).
>   - Site 1 BST power vary from -1dBm to +1dBm in steps of 0.5dB keeping the BST output powers at Site2, Site 3 at 0dBm.
>   - Choose the output power value at Site 1 BST which gives the best GSNR from above run.
>   - Lock the Site 1 BST value to above found value. Vary Site 2 BST output power from -1dBm to +1dBm in steps of 0.5dB and
>   keep  BST output power at Site3 at 0dBm.
>   - Choose the output power value at Site 2 BST which gives the best GSNR from above run.
>   - Lock the Site 1 BST and Site 2 BST value to above found values. Vary Site 3 BST output power from -1dBm to +1dBm in steps
>   of 0.5dB.
>   - In this manner, find the optimal launch power values for BSTs at Site 1, Site 2, Site 3.
>   - Now come back to Site 1 BST and vary its power value from -1dBm to +1dBm in steps of 0.5dB keeping the BST amps output
>   power Site 2, Site 3 at their optimal power value found from above runs.
>   - If Site 1 BST gives same output power value as before, then move to Site 2, Site 3 and check the same.
>   - If above run gives the same BST out power values at Site 1, Site 2,Site 3 then convergence has happened else repeat all the
>   above runs till a convergence is reached

if you are talking of **delta_** power_range_db defined in Span:
No each span has its own optimum power accounting for the already propagated power within the OMS. There is no global optimization with a feedback loop as you suggest. It is just one iteration.
 So,
- Site 1 BST power varies from -1dBm to +1dBm in steps of 0.5dB (delta_power_range_db) (compute the optimum from the simplified equation in target_power function in network.py). input power for the amp is the one set out of the Site1 ROADM (target_pch_out_db)
- Site 2 ILA power varies from -1dBm to +1dBm in steps of 0.5dB. input power for the amp is the result of Site1 booster gain (and delta_p and voa) setting computed in previous step.
- Site3 preamp power varies from -1dBm to +1dBm in steps of 0.5dB. input power for the amp is the result of Site2 ILA gain setting computed in previous step.
- Site3 booster power varies from -1dBm to +1dBm in steps of 0.5dB. input power for the amp is the one set out of the Site2 ROADM. 
-  ...
Please have a look to this paper , this clarifies what's in the current code  [J. -L. Auge, V. Curri and E. Le Rouzic, Open Design for Multi-Vendor Optical Networks, OFC 2019](https://ieeexplore.ieee.org/document/8696699), equation 4. See delta_power_range_db for more explaination.

**power_range_db** (from SI) is no used with path-request-run.

hope this helps

regards

**Reply by akhil-mutike** [reply]
Hi Esther(@EstherLerouzic ),
You have mentioned that power_range_db is not used for path-request-run. In this case, How does the Algo find the Optimal Reference Power?

**Reply by EstherLerouzic** [reply]
the parameter that is used for per span optimisation of delta power is **delta_power_range_db**  (in Span section of the library) https://gnpy.readthedocs.io/en/master/json.html#span. **power_range_db** (in SI) instead is used trigger a power sweep around and is only avalaible with the gnpy-transmission-example script https://gnpy.readthedocs.io/en/master/json.html#spectralinformation. Does this answer to your question ?

**Reply by akhil-mutike** [reply]
Thanks Esther(@EstherLerouzic ) for quick relpy,
According to  [J. -L. Auge, V. Curri and E. Le Rouzic, Open Design for Multi-Vendor Optical Networks, OFC 2019], we need to find first the optimal power for a reference span[20 Db]. So, we are skipping this step in path-request-run. My question is ,how good are the values obtained from only using delta_power_range_db( without the power sweep using the power_range_db)?  Should  we put a large range of values  in delta_power_range_db to get accurate power  values for different spans?

**Reply by EstherLerouzic** [reply]
Now I get your remark! The power sweep is only run  manually and does not change the design obtained with  gnpy-transmission-example script. So that for a given delta_power_range_db, both scripts result in the same design. This delta_power optimisation is (almost) independant from the power_dbm set in SI (except for reduction due to power saturation if total power+delta_p exceeds max power of amplifier).  The relative difference of power (delta_p) is computed based on a reference 20dB span. So increasing the range of delta_power_range_db will help to optimize spans delta_p that are very different from this reference, but will not give you the absolute optimum power. So assuming that power_dbm in SI is set to 0dBm and is the optimum power for a 20dB span, then all delta_ps in the network will be close to optimum. But if instead the power_dbm is not set according to this 20dB loss ref, than the absolute power will not be optimum. Does this answer your question ?

**Reply by akhil-mutike** [reply]
Thanks Esther(@EstherLerouzic ), Yeah i understand it now. What i understood is :

1. if we are using Path  Request  Run,  we need to provide correct(optimum) power_dbm for the reference span. so that delta_ps are optimum.
2.If we are using transmission example,  the script finds the optimum reference power by doing a power sweep around power_db and calculates optimum delta_ps with reference to optimum reference power.

Please correct me if I am wrong.

**Reply by EstherLerouzic** [reply]
point1 is correct and point 2 is almost correct except that this is not automatic: by setting the range in power_range_db: there is a computation of performance for each power of the range and the user can see which power give the best result (a list of performance computation is displayed on the CLI). delta_p are not recomputed. After the manual sweep, the user may want to change the power_dbm of the library, only in this case will the delta_p be recomputed (new design). 
Your series of questions shows that the documentation is not clear and that maybe the optimization process could be improved... Do you have suggestions ?

**Reply by akhil-mutike** [reply]
Hi Esther,
thank you for your confirmation.
The documentation definitely requires  improvement.
In the paper [J. -L. Auge, V. Curri and E. Le Rouzic, Open Design for Multi-Vendor Optical Networks, OFC 2019](https://ieeexplore.ieee.org/document/8696699), the authors have mentioned that power sweep need not be done for all OMS sections, if it satisfies these two conditions:

1.  uniform span vs ROADM distribution across the network (i.e. ROADM noise contribution is consistent) 
2.  Fine power differentiation / span so that the result of the power sweep Prefopt=Cte

I am unable to understand the first point, can you please help me with this.
Second point tells us basically have fine power differentiation(delta power=[x,y,z], Z should be smaller).what about x and y .How do you determine optimum  x nad y?

**Comment by SaiBhyri** [comment]
Thanks Esther.

Regards,
Sai

**Comment by Liuyuan0803** [comment]
Hi Esther,
Thanks for your answer, it helps me to understand these parameters a lot. But I have one more question.
In your document, the description of **f_max** and **f_min** is that  _In Hz. Define spectrum boundaries. Note that due to backward compatibility, the first channel central frequency is placed at f_min + spacing and the last one at f_max_. According to your formula of the channel number: (**f_max** - **f_min**)/(**spacing**),  if the first and the last channel central frequency are placed at f_min + (spacing/2) and f_max - (spacing/2) is more suitable?
